api/cpp/esp-idf/slint/src/slint-esp.cpp

```diff
@@ -214,43 +214,71 @@ void EspPlatform::run_event_loop()
+                            xSemaphoreTake(sem_vsync_end, portMAX_DELAY);
 #endif
 
-                        // Assuming that using double buffer means that the buffer comes from the
-                        // driver and we need to pass the exact pointer.
-                        // https://github.com/espressif/esp-idf/blob/53ff7d43dbff642d831a937b066ea0735a6aca24/components/esp_lcd/src/esp_lcd_panel_rgb.c#L681
-                        esp_lcd_panel_draw_bitmap(panel_handle, 0, 0, size.width, size.height,
-                                                  buffer1.data());
-                        std::swap(buffer1, buffer2.value());
-                    } else {
-                        for (int y = o.y; y < o.y + s.height; y++) {
-                            for (int x = o.x; x < o.x + s.width; x++) {
-                                // Swap endianess to big endian
-                                auto px =
-                                        reinterpret_cast<uint16_t *>(&buffer1[y * size.width + x]);
-                                *px = (*px << 8) | (*px >> 8);
+                            // Assuming that using double buffer means that the buffer comes from
+                            // the driver and we need to pass the exact pointer.
+                            // https://github.com/espressif/esp-idf/blob/53ff7d43dbff642d831a937b066ea0735a6aca24/components/esp_lcd/src/esp_lcd_panel_rgb.c#L681
+                            esp_lcd_panel_draw_bitmap(panel_handle, 0, 0, size.width, size.height,
+                                                      buffer1.data());
+
+                            std::swap(buffer1, buffer2.value());
+                        } else {
+                            for (int y = o.y; y < o.y + s.height; y++) {
+                                for (int x = o.x; x < o.x + s.width; x++) {
+                                    // Swap endianess to big endian
+                                    auto px = reinterpret_cast<uint16_t *>(
+                                            &buffer1[y * size.width + x]);
+                                    *px = (*px << 8) | (*px >> 8);
+                                }
+                                esp_lcd_panel_draw_bitmap(panel_handle, o.x, y, o.x + s.width,
+                                                          y + 1,
+                                                          buffer1.data() + y * size.width + o.x);
                             }
-                            esp_lcd_panel_draw_bitmap(panel_handle, o.x, y, o.x + s.width, y + 1,
-                                                      buffer1.data() + y * size.width + o.x);
                         }
                     }
+                } else {
+                    slint::platform::Rgb565Pixel *lb =
+                            (slint::platform::Rgb565Pixel *)heap_caps_malloc(
+                                    size.width * 2, MALLOC_CAP_INTERNAL | MALLOC_CAP_8BIT);
```

Comment:
Is there a way to use RAII here?
`std::unique_ptr<slint::platform::Rgb565Pixel[], decltype(&free)>` maybe?

Reply:
Good idea! That should be possible.